test/selectors/dashboard-selectors_spec.js

```diff
@@ -60,23 +64,44 @@ describe('dashboard selectors', () => {
     })
   })
 
-  describe('getSortedStudents', () => {
-    describe('when sortBy = "name"', () => {
-      const state = fromJS({
-        report: {
-          students: { 1: s1, 2: s2, 3: s3 }
-        },
-        dashboard: {
-          sortBy: 'name'
+  describe('getStudentTotalProgress', () => {
+    it('should return hash with student total progress', () => {
+      expect(getStudentTotalProgress(state({})).toJS()).to.eql(
+        {
+          1: 1.5,
+          2: 1.5,
+          3: 0
         }
-      })
+      )
+    })
+  })
 
+  describe('getSortedStudents', () => {
+    describe('when sorting by name', () => {
       it('should return sorted list of students', () => {
-        expect(getSortedStudents(state).toJS()).to.eql(
+        expect(getSortedStudents(state({})).toJS()).to.eql(
           // Students sorted by name (last name first)
           [ s3, s2, s1 ]
         )
       })
     })
+
+    describe('when sorting by most progress', () => {
+      it('should return sorted list of students', () => {
+        expect(getSortedStudents(state({sortBy: SORT_BY_MOST_PROGRESS})).toJS()).to.eql(
```

Comment:
The ability to stub out selectors would really help here. Since I already test `getStudentTotalProgress` above, it would be nice to just mock its output here. Right now, this test does too much.

Reply:
If you really want, note that you could create named function that is then passed to `createSelector`. Something like this:

```javascript
const getSortedStudents = (students, totalProgress) => { .... }
const getSortedStudentsSelector = createSelector([getStudents, getTotalProgress], getSortedStudents)
```
Export both, and you could test only test `getSortedStudents`. Then it's super easy to provide any argument you want, without providin the whole state.

However, as long as the test is pretty concise, readable, and setup isn't too bad, I'd say it's an advantage that we test more than less. Generally, I'm not a fan of tiny unit tests that stub and mock everything and test some tiny logic. I find that it's easy to create tests that actually test themselves, not the application anymore. :wink: But there's a thin line here. I guess that in case of some selectors, this approach can be helpful. It's a bit overhelming that you need to create the whole app state. Maybe even here it would make sense, but since the state-crafting work has been done already, so I wouldn't change it.